### PR TITLE
CVSB-9489 Forcing the user to update the app if there is a newer breaking version

### PR DESCRIPTION
**Pull request by agoncearenco**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -135,7 +136,9 @@
       "cordova-plugin-ms-adal": {},
       "cordova-plugin-firebase": {},
       "cordova-plugin-network-information": {},
-      "cordova-plugin-secure-storage": {}
+      "cordova-plugin-secure-storage": {},
+      "cordova-plugin-app-version": {},
+      "cordova-plugin-exit": {}
```
Comment (KHodderDVSA) [inline package.json]: no dependency for this one?

File: src/providers/global/sync.service.spec.ts [hunk 2]
```diff
@@ -87,4 +96,30 @@ describe('Provider: SyncService', () => {
     syncService.handleData(myArray);
     expect(events.publish).toHaveBeenCalled();
   });
+
+  it('should show the update popup if the app version is not the latest and there is no current visit', () => {
```
Comment (KHodderDVSA) [inline src/providers/global/sync.service.spec.ts]: Extra test for if new version but breaking = "false"

File: src/providers/global/sync.service.ts [hunk 3]
```diff
@@ -57,6 +63,42 @@ export class SyncService {
...
+    promises.push(this.storageService.read(STORAGE.VISIT));
+
+    try{
+      let results = await Promise.all(promises);
+      const currentAppVersion = results[0];
+      const latestAppVersionModel: AppVersionModel = results[1].body['mobile-app'];
+      const visit = results[2];
+      if(currentAppVersion !== latestAppVersionModel.version && latestAppVersionModel.breaking === 'true' && !visit){
+        this.createUpdatePopup().present();
+        return Promise.resolve(true);
```
Comment (KHodderDVSA) [inline src/providers/global/sync.service.ts]: return is scoped to if statement. no return or resolve(false) if no update required?

**Comment by vladddragomir** [comment]
besides @KHodderDVSA's comment regarding the extra unit test (which i agree), i'm ok with the pr

File: package.json (hunk 1 above)
Comment (agoncearenco) [reply package.json]: You're right, I might have reverted it by accident.

File: src/providers/global/sync.service.spec.ts (hunk 2 above)
Comment (agoncearenco) [reply src/providers/global/sync.service.spec.ts]: I'm removing the breaking flag check. The ACs did not specify to verify if the change is breaking or not

Comment (agoncearenco) [reply src/providers/global/sync.service.spec.ts]: Will update the tests

File: src/providers/global/sync.service.ts (hunk 3 above)
Comment (agoncearenco) [reply src/providers/global/sync.service.ts]: I do not use this return value. Will remove all the returns.

File: src/providers/global/sync.service.spec.ts (hunk 2 above)
Comment (agoncearenco) [reply src/providers/global/sync.service.spec.ts]: Requested change completed.

File: src/providers/global/sync.service.ts (hunk 3 above)
Comment (agoncearenco) [reply src/providers/global/sync.service.ts]: Requested change completed.

File: package.json (hunk 1 above)
Comment (agoncearenco) [reply package.json]: Requested change completed.

File: src/providers/global/sync.service.ts [hunk 4]
```diff
@@ -63,7 +63,7 @@ export class SyncService {
     this.getAllData();
   }
 
-  public async isUpdateNeeded(): Promise<boolean>{
+  public async isUpdateNeeded(){
```
Comment (KHodderDVSA) [inline src/providers/global/sync.service.ts]: no longer a boolean returning function (I approve), so the name is ... misleading. Perhaps "checkForUpdate" would be more appropriate.

Comment (agoncearenco) [reply src/providers/global/sync.service.ts]: Right! will change

Comment (agoncearenco) [reply src/providers/global/sync.service.ts]: Requested change completed.

**Pull request merged**